lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
Do we really need to use the `Retraction` class as a mediator between `Contact` model and `Contact` federation entity? Can't we map `Contact` model directly to `Contact` entity and pass the `Contact` model instance to the dispatcher instead of using `Retraction` for that?

In the current implemenation you have to keep "Contact" type as a special case everywhere you deal with the retraction. And I think the original point was that we don't use the `Retraction` entity for the stop sharing event, so I don't think `Contact` entity dispatching has anything to do with retractions anymore.